packages/java-parser/scripts/unicode.js

```diff
@@ -150,11 +150,20 @@ function readUnicodeData() {
 // Generating a unicodesets.js file so that we don't have to reparse the file each time the parser is called.
 function generateFile() {
   let data = `
-  /*File generated with unicode.js*/
+  /*File generated with ../scripts/unicode.js using ../resources/Unicode/UnicodeData.txt.
+  * As Java Identifiers may contains unicodes letters, this file defines two sets of unicode
+  * characters, firstIdentChar used to help to determine if a character can be the first letter
+  * of a JavaIdentifier and the other one (restIdentChar) to determine if it can be part of a
+  * JavaIdentifier other than the first character.
+  * Java uses the same file UnicodeData.txt as the unicode.js script to define the unicodes.
+  * For more:
+  *   https://github.com/jhipster/prettier-java/issues/116
+  *   https://github.com/jhipster/prettier-java/pull/155
+  */
   "use strict"
-  const f = (o, a) => {
-    a.forEach(e => {
-      [...Array(e[1] - e[0] + 1).keys()].map(i => o.add(i + e[0]));
+  function addRanges(o, arrays) {
```

Comment:
note that a simpler imperative approach would be much faster.
```javascript
const f = (set, rangesArr) => {
  for (let i = 0; i < rangesArr.length; i++) {
    const range = rangesArr[i];
    const start =  range[0];
    const end = range[1];
    for (let codePoint = start; codePoint <= end; codePoint++) {
      set.add(codePoint)
    }
  }
};
```

And arguably easier to understand as you are performing multiple transformations in a single line.
- https://youtu.be/ecekSCX3B4Q?t=1667

Reply:
I will keep that in mind :).